salt/runners/survey.py

```diff
@@ -171,7 +171,7 @@ def _get_pool_results(*args, **kwargs):
 
     # hash minion return values as a string
     for minion in sorted(minions):
-        digest = hashlib.sha256(str(minions[minion])).hexdigest()
+        digest = hashlib.sha256(str(minions[minion]).encode('utf-8')).hexdigest()  
```

Comment:
We should always be using `__salt_system_encoding__` instead of assuming `utf-8`.